src/components/Modal/AddName/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import {
+	ButtonsWrapper,
+	CancelButton,
+	CloseIconDiv,
+	Input,
+	InputWrap,
+	Label,
+	ModalTitle,
+	Wrapper,
+} from "./styles";
+import { ErrorMessage, StartButton } from "../../CategoryCard/styles";
+import { useQuizData } from "../../../context/quizContext";
+import { CLOSE__MODAL, SET__PLAYER__NAME } from "../../../constants";
+import { ReactComponent as CloseIcon } from "../../../icons/close-icon.svg";
+import { useHistory } from "react-router";
+
+function AddName() {
+	const { dispatch } = useQuizData();
+	const [name, setName] = useState("");
+	const [isNameActive, setIsNameActive] = useState(false);
+	const [message, setMessage] = useState("");
+	const history = useHistory();
+	const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setName(e.target.value);
+		if (e.target.value.length > 0) {
+			return setIsNameActive(true);
+		}
+		return setIsNameActive(false);
+	};
+
+	const modalCloseHandler = () => {
+		return dispatch({ type: CLOSE__MODAL });
+	};
+
+	const addNameHandler = () => {
+		if (name.length === 0) {
+			return setMessage("Please Enter a name");
+		}
+		dispatch({ type: SET__PLAYER__NAME, payload: name });
+		dispatch({ type: CLOSE__MODAL });
+		return history.push("/quiz");
```

Comment:
Instead of using `history`, use `useNavigate()`
```suggestion
		return navigate("/quiz");
```

Reply:
I not using react-router-dom v6. So useNavigate hook is not there.